apps/vscode/vscode.win.talon

```diff
@@ -0,0 +1,235 @@
+# Microsoft - Visual Studio Code
+# see app/vscode.talon for voice commands
+
+os: windows
+app: Visual Studio Code
+app: Code.exe
+-
+
+
+tag(): git
+# TODO When tags are supported
+# tag(): ide
+
+# General
+action(user.ide_command_palette):
+  key(ctrl-shift-p)
+
+action(app.tab_next):
+  key(ctrl-k)
+  key(ctrl-pagedown)
+  
+action(app.tab_previous):
+  key(ctrl-k)
+  key(ctrl-pageup)
+  
+# Folding
+action(user.ide_expand_deep):
+  key(ctrl-k ctrl-])
+action(user.ide_expand_all):
+  key(ctrl-k ctrl-j)
+action(user.ide_expand_region):
+  key(ctrl-shift-])
+action(user.ide_collapse_deep):
+  key(ctrl-k ctrl-[)
+action(user.ide_collapse_all):
+  key(ctrl-k ctrl-0)
+action(user.ide_collapse_region):
+  key(ctrl-shift-[)
+
+# Splits
+action(user.ide_split_right):
+  user.ide_command_palette()
+  insert("workbench.action.splitEditorRight")
+  key(enter)
+
+action(user.ide_split_left):
+  user.ide_command_palette()
+  insert("workbench.action.splitEditorLeft")
+  key(enter)
+
+action(user.ide_split_up):
+  user.ide_command_palette()
+  insert("workbench.action.splitEditorUp")
+  key(enter)
+
+action(user.ide_split_down):
+  user.ide_command_palette()
+  insert("workbench.action.splitEditorDown")
+  key(enter)
+
+#Refactor
+
+action(user.ide_refactor):
+  key(ctrl-a)
+  # Format Document
+  key(alt-shift-f)
+  key(ctrl-s)
+
+action(user.ide_refactor_in_line):
+  # Format Document
+  key(alt-shift-f)
+  #TODO are we happy to save everytime ?
+  key(ctrl-s)
+
+action(user.ide_refactor_rename):
+  # Rename Symbol
+  key(f2)
+
+# Navigate
+action(user.ide_follow):
+  # Go to Definition
+  key(f12)
+
+action(user.ide_go_back):
+  key(alt-left)
+
+action(user.ide_go_forward):
+  key(alt-right)
+
+action(user.ide_up_cursor):
```

Comment:
The ide_*_cursor actions are intended for adding multi-select cursors, not for selections. 

See this: https://code.visualstudio.com/docs/editor/codebasics

e.g., from jetbrains
```
action(user.ide_multi_cursor): key(shift-alt-insert)
action(user.ide_multi_cursor_stop): key(escape)
action(user.ide_up_cursor): user.idea("action EditorCloneCaretAbove")
action(user.ide_down_cursor): user.idea("action EditorCloneCaretBelow")
```